Assets/ZSSRichTextEditor.js

```diff
@@ -806,8 +806,446 @@ ZSSEditor.removeImage = function(imageNodeIdentifier) {
     }
 };
 
+/**
+ *  @brief      Updates the currently selected image, replacing its markup with
+ *  new markup based on the specified meta data string.
+ *
+ *  @param      imageMetaString   A JSON string representing the updated meta data.
+ */
+ZSSEditor.updateCurrentImageMeta = function( imageMetaString ) {
+    //    ZSSEditor.restoreRange();
+
+    if ( ZSSEditor.currentEditingImage ) {
+        var imageMeta = JSON.parse( imageMetaString );
+        var html = ZSSEditor.createImageFromMeta( imageMeta );
+
+        ZSSEditor.selectImageCaptionNode( ZSSEditor.currentEditingImage );
+        ZSSEditor.insertHTML( html );
+        return; // return to avoid sending enabled styles twice.
+    }
+
+    ZSSEditor.sendEnabledStyles();
+}
+
+
+/**
+ *  @brief      Selects the specified image, and its anchor and caption shortcode (if any).
+ *
+ *  @param      imageNode   An image node in the DOM to inspect.
+ */
+ZSSEditor.selectImageCaptionNode = function( imageNode ) {
+    var node = imageNode;
+    if ( node.parentNode && node.parentNode.nodeName === 'A' ) {
+        node = node.parentNode;
+    }
```

Comment:
Each if block should be checked independently. An image may or may not have an anchor, or a caption. This is kind of explained in the brief but maybe it could be clarified.